src/tests/simulation/src/main.rs

```diff
@@ -294,51 +294,70 @@ async fn kill_node() {
         tracing::info!("kill {name}");
         madsim::runtime::Handle::current().kill(&name);
     }
-    tokio::time::sleep(std::time::Duration::from_secs(1)).await;
+    tokio::time::sleep(Duration::from_secs(1)).await;
     for name in &nodes {
         tracing::info!("restart {name}");
         madsim::runtime::Handle::current().restart(&name);
     }
-    tokio::time::sleep(std::time::Duration::from_secs(10)).await;
 }
 
 #[cfg(not(madsim))]
 async fn kill_node() {}
 
-struct HookImpl;
-
-#[async_trait::async_trait]
-impl sqllogictest::Hook for HookImpl {
-    async fn on_stmt_complete(&mut self, _sql: &str) {
-        kill_node().await;
-    }
-
-    async fn on_query_complete(&mut self, _sql: &str) {
-        kill_node().await;
-    }
-}
-
 async fn run_slt_task(glob: &str, host: &str) {
     let risingwave = Risingwave::connect(host.to_string(), "dev".to_string()).await;
-    if ARGS.kill_compute {
+    if ARGS.kill_compute || ARGS.kill_meta {
         risingwave
             .client
             .simple_query("SET RW_IMPLICIT_FLUSH TO true;")
             .await
             .expect("failed to set");
     }
     let mut tester = sqllogictest::Runner::new(risingwave);
-    tester.set_hook(HookImpl);
     let files = glob::glob(glob).expect("failed to read glob pattern");
     for file in files {
         let file = file.unwrap();
         let path = file.as_path();
         println!("{}", path.display());
-        tester
-            .run_file_async(path)
-            .await
-            .map_err(|e| panic!("{e}"))
-            .unwrap()
+        for record in sqllogictest::parse_file(path).expect("failed to parse file") {
+            if let sqllogictest::Record::Halt { .. } = record {
+                break;
+            }
+            let is_create = matches!(&record, sqllogictest::Record::Statement { sql, .. } 
+                if sql.starts_with("CREATE") || sql.starts_with("create"));
+            let is_drop = matches!(&record, sqllogictest::Record::Statement { sql, .. } 
+                if sql.starts_with("DROP") || sql.starts_with("drop"));
+            let dont_kill = matches!(&record, sqllogictest::Record::Statement { sql, .. } 
+                if sql.starts_with("INSERT") || sql.starts_with("UPDATE") || sql.starts_with("FLUSH")
```

Comment:
Also don't kill on `delete`?

Reply:
What if a sql contains both upper and lower case is added by mistake? It's hard to be found in review after all. 😭

Reply:
Or simply `sql.to_lower_case().starts_with(..)` 🥵. BTW, is SQL parsed in sqllogictest already trimmed?

Reply:
Added `trim_start` and `to_lower_case` for these cases. 🥵

> BTW, is SQL parsed in sqllogictest already trimmed?

No. It will keep the original string.